src/Forms/Form.php

```diff
@@ -335,6 +335,18 @@ public function addConfirmation($message)
         return $this;
     }
 
+    /**
+     * setOnKeyDown
+     * Add a javascript function to the form's onkeydown event.
+     * @param string $function
+     * @return self
+     */
+    public function setOnKeyDown($function)
```

Comment:
Thanks for making these changes. I think this method would also work best as `enableAutoSave(string $url, string $formId)`, with the `$keydownJS` being created and handled internally in the method. That way, if the JS logic or events it's bound to ever change, the implementation would only need to change in one place.